src/test/regress/expected/multi_metadata_sync.out

```diff
@@ -1279,8 +1293,10 @@ DROP TABLE mx_table;
 \c - postgres - :master_port
 INSERT INTO pg_dist_placement SELECT * FROM pg_dist_placement_temp;
 INSERT INTO pg_dist_partition SELECT * FROM pg_dist_partition_temp;
+INSERT INTO citus.pg_dist_object SELECT * FROM pg_dist_object_temp ON CONFLICT ON CONSTRAINT pg_dist_object_pkey DO NOTHING;
 DROP TABLE pg_dist_placement_temp;
 DROP TABLE pg_dist_partition_temp;
+DROP TABLE pg_dist_object_temp;
```

Comment:
It seems out DROP hypothesis doesn't hold for all the types. When we drop some distributed objects from the coordinator, they remain on the workers. See the example for sequences:
```SQL
CREATE SEQUENCE my_seq;
CREATE TABLE dist_table(a int, b int DEFAULT nextval('my_seq'), c bigserial);
SELECT create_distributed_table('dist_table', 'a');

select  pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object ;
       pg_identify_object_as_address       
-------------------------------------------
 (role,{onderkalaci},{})
 (database,{postgres},{})
 (schema,{public},{})
 (sequence,"{public,my_seq}",{})
 (sequence,"{public,dist_table_c_seq}",{})
(5 rows)


-- on the worker
select  objid, pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid != 19202;
 objid |       pg_identify_object_as_address       
-------+-------------------------------------------
    10 | (role,{onderkalaci},{})
 13720 | (database,{postgres},{})
  2200 | (schema,{public},{})
 19200 | (sequence,"{public,my_seq}",{})
 19251 | (sequence,"{public,dist_table_c_seq}",{})
(5 rows)


-- back on the coordinator
DROP TABLE dist_table;
select  pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object ;
  pg_identify_object_as_address  
---------------------------------
 (role,{onderkalaci},{})
 (database,{postgres},{})
 (schema,{public},{})
 (sequence,"{public,my_seq}",{})
(4 rows)

Time: 0.498 ms


-- on the worker
select  pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object;
ERROR:  cache lookup failed for relation 19251
Time: 0.539 ms

-- we know that 19251 == (sequence,"{public,dist_table_c_seq}",{})
```

Not sure yet if this only relevant sequences. For example, a type example works fine (which might be good to have in the test):
```SQL
CREATE TYPE tt1 as (a int);
CREATE TABLE dist_table(a int, b tt1);
SELECT create_distributed_table('dist_table', 'a');

-- on both the worker and coordinator
-- but objid from the worker for this example 
select  objid, pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object;
 objid | pg_identify_object_as_address 
-------+-------------------------------
    10 | (role,{onderkalaci},{})
 13720 | (database,{postgres},{})
  2200 | (schema,{public},{})
 16957 | (type,{public.tt1},{})
(4 rows)


drop type tt1 cascade;
NOTICE:  drop cascades to column b of table dist_table
DROP TYPE
Time: 35.539 ms

-- on both the worker and coordinator
-- works as expected
select  objid, pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object;
 objid | pg_identify_object_as_address 
-------+-------------------------------
    10 | (role,{onderkalaci},{})
 13720 | (database,{postgres},{})
  2200 | (schema,{public},{})
(3 rows)
```

I think a thorough `drop` test would be very nice to have. Drop dist objects and make sure that they are dropped from the workers as well.